encodings/fastlanes/src/for/pipeline.rs

```diff
@@ -0,0 +1,352 @@
+        self
+    }
+
+    fn vtype(&self) -> VType {
+        VType::Primitive(self.ptype)
+    }
+
+    fn children(&self) -> &[Arc<dyn Operator>] {
+        &self.child
+    }
+
+    fn with_children(&self, mut children: Vec<Arc<dyn Operator>>) -> Arc<dyn Operator> {
+        assert_eq!(children.len(), 1);
+        Arc::new(FoROperator {
+            child: [children.remove(0)],
+            reference: self.reference.clone(),
+            ptype: self.ptype,
+            encoded_ptype: self.encoded_ptype,
+        })
+    }
+
+    fn in_place(&self) -> bool {
+        true
+    }
+
+    fn bind(&self, ctx: &dyn BindContext) -> VortexResult<Box<dyn Kernel>> {
+        let VType::Primitive(ptype) = self.vtype() else {
+            vortex_bail!("FoROperator only supports primitive types");
+        };
+
+        match_each_integer_ptype!(ptype, |T| {
+            match_each_integer_ptype!(self.encoded_ptype, |E| {
+                Ok(Box::new(FoRKernel::<T, E> {
+                    child: ctx.children()[0],
+                    reference: self
+                        .reference
+                        .as_primitive()
+                        .typed_value::<T>()
+                        .vortex_expect("reference value not of type T"),
+                    _marker: PhantomData,
+                }))
+            })
+        })
+    }
+
+    fn reduce_parent(&self, parent: Arc<dyn Operator>) -> Option<Arc<dyn Operator>> {
+        let compare = parent.as_any().downcast_ref::<ScalarCompareOperator>()?;
+
+        let new_ref = match_each_native_ptype!(self.reference.as_primitive().ptype(), |P| {
+            let compare = compare
+                .scalar
+                .as_primitive()
+                .typed_value::<P>()
+                .vortex_expect("must have ptype");
+            let reference = self
+                .reference
+                .as_primitive()
+                .typed_value::<P>()
+                .vortex_expect("must have ptype");
+            // TODO: handle overflow
+            Scalar::from(compare - reference)
+        });
+
+        Some(Arc::new(ScalarCompareOperator::new(
+            self.children()[0].clone(),
+            compare.op,
+            new_ref,
+        )))
+    }
+}
+
+// TODO(ngates): we should try putting the const bit width as a generic here, to avoid
+//  a switch in the fastlanes library on every invocation of `unchecked_unpack`.
+pub(crate) struct FoRKernel<T: NativePType, E: NativePType> {
+    child: VectorId,
+    reference: T,
+    _marker: PhantomData<E>,
+}
+
+impl<T, E> Kernel for FoRKernel<T, E>
+where
+    T: NativePType + Element + WrappingAdd,
+    E: NativePType + Element,
+{
+    fn seek(&mut self, _chunk_idx: usize) -> VortexResult<()> {
+        Ok(())
+    }
+
+    fn step(
+        &mut self,
+        ctx: &dyn KernelContext,
+        selected: BitView,
+        out: &mut ViewMut,
+    ) -> Poll<VortexResult<()>> {
+        let vec = ctx.vector(self.child);
```

Comment:
You say you're in-place, but then we take a child? Does the planner ignore in-place for now?